library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInputOperator.java

```diff
@@ -361,14 +361,22 @@ public String queryToRetrieveData()
   public void activate(Context.OperatorContext context)
   {
     for (int i = 0; i < columnDataTypes.size(); i++) {
-      final int type = columnDataTypes.get(i);
+      final int type;
+      try {
+        type = columnDataTypes.get(i);
+      }
+      catch (NullPointerException e) {
+        throw new RuntimeException("Extra fieldname " +
```

Comment:
Explore the possibility of  handling this scenatio in  populateColumnDataTypes() instead of expecting an exception